syntax = "proto3";

option go_package = "sensor";
option cc_enable_arenas = true;

import "internalapi/sensor/collector.proto";
import "internalapi/sensor/network_connection_info.proto";
import "api/v1/empty.proto";

package sensor;

message NetworkConnectionInfoMessage {
    oneof msg {
        CollectorRegisterRequest register = 1;
        NetworkConnectionInfo info = 2;
    }
}

message NetworkFlowsControlMessage {
    IPAddressList public_ip_addresses = 1;
}

message IPAddressList {
    repeated fixed32 ipv4_addresses = 1;
}

// A Sensor service that allows Collector to send NetworkConnectionInfo messages
service NetworkConnectionInfoService {
    // Note: the response is a stream due to a bug in the C++ GRPC client library. The server is not expected to
    // send anything via this stream.
    rpc PushNetworkConnectionInfo (stream NetworkConnectionInfoMessage) returns (stream NetworkFlowsControlMessage);
}
